invisible_cities/cities/components.py

```diff
@@ -589,6 +594,31 @@ def MC_hits_from_files(files_in : List[str], rate: float) -> Generator:
                        timestamp    = timestamp(evt))
 
 
+def dhits_from_files(paths: List[str]) -> Iterator[Dict[str,Union[HitCollection, pd.DataFrame, MCInfo, int, float]]]:
+    """Reader of the files, yields HitsCollection, pandas DataFrame with
+    run_number, event_number and timestamp."""
+    for path in paths:
+        try:
+            dhits_df = load_dst (path, 'DECO', 'Events')
+        except tb.exceptions.NoSuchNodeError:
+            continue
+
+        with tb.open_file(path, "r") as h5in:
+            try:
+                run_number  = get_run_number(h5in)
+            except (tb.exceptions.NoSuchNodeError, IndexError):
+                continue
+
+            event_info = load_dst(path, 'Run', 'events')
+            for evt in dhits_df.event.unique():
+                timestamp = event_info[event_info.evt_number==evt].timestamp.values[0]
+                dhits = hits_from_df(dhits_df.loc[dhits_df.event==evt])
```

Comment:
This is not correct since the DECO hits for a single event can contain multiple peaks. Since no peak information is included in the final track information output, we can not directly check if a multiple-track event comes from a real spliting or by a multiple-S2 event. This is not a relevant issue as long as commonly used 1S2 and 1-track cut are performed, but it is biasing the efficiencies.

This PR also modifies the search of track extremes in order to be deterministic but it does not adapt the tests, also affecting esmeralda tests and making the PR more difficult to review. I think this modifications should be a PR appart. I will do a different PR implementing the same functionality of this city, essentially running paolina algorithm over deconvoluted hits, without modifying the track extreme search.

Reply:
Sorry, I don't understand completely your plan. If you want to add a new PR that exclusively adds the city to run paolina over deconvoluted hits, what would you do with this PR? just ignore it? would you use this just for solving the extreme issue? There is another thing to take into account, this PR apart from doing both things, also moves some functions from `Esmeralda` to `components.py`, would you also add that functionality in your new PR?

In my opinion I believe that the best approach could be removing all the extreme-related issue from this PR, and then add that new PR that you commented to take it into account. So the goal of this PR will keep being to add the tracking city.

Reply:
Yes, as you might see in #798 I just did another PR since I didn't aim to undo your changes related with the extreme finder. I didn't move the functions still from `esmeralda.py` to ease the review, but I intend to do it once the PR is reviewed. Then the main difference with this PR is that I left the paolina functions unmodified, which in my opinion should be done, if ever, in a different PR. It also changes the source to solve the issue with multipeak events, such that those events are filtered.